preprocessors/openstreetmap/osm_service.py

```diff
@@ -174,193 +177,144 @@ def process_streets_data(OSM_data, bbox_coordinates):
-        # needed.
-
-        elif i > 1 and j > i:
-            # Get the index of the last element(node) in the restricted list
-            # (node_list) from the unbounded list (node_list1).
-            index = node_list1.index(node_list[i - 1])
-            if index < j - 1:  # If true, the last node
-                # in node_list has a succeeding node in the original node list
-                # (node_list1), so compute for the succeeding node.
-                index = index + 1
-                lat1 = node_list[i - 1]["lat"]
-                lon1 = node_list[i - 1]["lon"]
-                lat2 = node_list1[index]["lat"]
-                lon2 = node_list1[index]["lon"]
-                a = (lat1, lon1)
-                b = (lat2, lon2)
-                result = Geodesic.WGS84.Inverse(*a, *b)
-                # Bearing between the single point node and the succeeding node
-                # in degrees
-                bearing = result["azi1"]
-                node_params = {
-                    "id": node_list1[index]["id"],
-                    "lat1": lat1,
-                    "lon1": lon1,
-                    "lat2": lat2,
-                    "lon2": lon2
-                }
-                succeeding_node = compute_new_node(
-                    node_params, bearing, bbox_coordinates)
-                node_list.append(succeeding_node)
-
-            # Get the index of the first element(node) in the bounded list
-            # (node_list)from the unbounded list (node_list1)
-            index = node_list1.index(node_list[0])
-            if index > 0:  # If true, the first node
-                # in node_list has a preceding node in the original node list
-                # (node_list1), so, compute for the preceding node.
-                index = index - 1
-                lat1 = node_list[0]["lat"]
-                lon1 = node_list[0]["lon"]
-                lat2 = node_list1[index]["lat"]
-                lon2 = node_list1[index]["lon"]
-                a = (lat1, lon1)
-                b = (lat2, lon2)
-                result = Geodesic.WGS84.Inverse(*a, *b)
-                bearing = result["azi1"]
-                # Bearing between the single point node and the succeeding node
-                # in degrees
-                node_params = {
-                    "id": node_list1[index]["id"],
-                    "lat1": lat1,
-                    "lon1": lon1,
-                    "lat2": lat2,
-                    "lon2": lon2
-                }
-                preceding_node = compute_new_node(
-                    node_params, bearing, bbox_coordinates)
-                node_list.insert(0, preceding_node)
-    return node_list
-
-# Create a new node
+def get_new_nodes(bounded_nodes, unbounded_nodes, bbox_coordinates):
+    # - bounded_nodes is a list of only nodes of a street that fall within
+    # the bounding box.
+    # - unbounded_nodes on the other hand has all the nodes of a street.
+    i = len(bounded_nodes)
```

Comment:
Same comment about arbitrary variable names.

Reply:
Issue raised on the use of arbitrary variable names is now addressed.